core/src/main/java/com/graphhopper/routing/weighting/custom/PriorityCustomConfig.java

```diff
@@ -0,0 +1,170 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.weighting.custom;
+
+import com.graphhopper.routing.profiles.*;
+import com.graphhopper.routing.util.CustomModel;
+import com.graphhopper.util.EdgeIteratorState;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.prep.PreparedGeometryFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+final class PriorityCustomConfig {
+    private List<EdgeToValueEntry> priorityList = new ArrayList<>();
+
+    public PriorityCustomConfig(CustomModel customModel, EncodedValueLookup lookup) {
+        add(lookup, customModel.getVehicleWeight(), "vehicle_weight", MaxWeight.KEY);
+        add(lookup, customModel.getVehicleWidth(), "vehicle_width", MaxWidth.KEY);
+        add(lookup, customModel.getVehicleHeight(), "vehicle_height", MaxHeight.KEY);
+        add(lookup, customModel.getVehicleLength(), "vehicle_length", MaxLength.KEY);
+
+        for (Map.Entry<String, Object> entry : customModel.getPriority().entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+
+            if (value == null) {
+                throw new IllegalArgumentException("Missing value for " + key + " in 'priority'");
+            } else if (value instanceof Number) {
+                if (key.startsWith(GeoToValueEntry.key(""))) {
+                    Geometry geometry = GeoToValueEntry.pickGeometry(customModel, key);
+                    priorityList.add(new GeoToValueEntry(new PreparedGeometryFactory().create(geometry), ((Number) value).doubleValue(), 1));
+                } else {
+                    BooleanEncodedValue encodedValue = getEV(lookup, "priority", key, BooleanEncodedValue.class);
+                    priorityList.add(new BooleanToValueEntry(encodedValue, ((Number) value).doubleValue(), 1));
+                }
+            } else if (value instanceof Map) {
+                EnumEncodedValue enumEncodedValue = getEV(lookup, "priority", key, EnumEncodedValue.class);
+                Enum[] enumValues = enumEncodedValue.getValues();
+                double[] values = createEnumToDoubleArray("priority." + key, 1, 0, 100,
+                        enumValues, (Map<String, Object>) value);
+                normalizeFactor(values, 1);
+                priorityList.add(new EnumToValueEntry(enumEncodedValue, values));
+            } else {
+                throw new IllegalArgumentException("Type " + value.getClass() + " is not supported for " + key + " in 'priority'");
+            }
+        }
+    }
+
+    static <T extends EncodedValue> T getEV(EncodedValueLookup lookup, String name, String key, Class<T> encodedValueType) {
+        if (!lookup.hasEncodedValue(key))
+            throw new IllegalArgumentException("Cannot find '" + key + "' specified in '" + name + "'");
```

Comment:
It might be useful to change this to `Cannot find encoded value' " + key...` ? And even better then list the available encoded values? The latter requires us to be able to get *all* encoded values from `EncodedValueLookup` (or at least their names like `toEncodedValuesAsString` in `EncodingManager`, but I think this should be possible either way.

Reply:
Hmmh, ok, I will add this as we already need it for InfoResource.

Reply:
I think its useful anyway no? If EncodingManager (or EncodedValueLookup) is supposed to be a 'collection of encoded values` there should be no reason not to be able to obtain the encoded values (all of them, one of them, one after the other etc.). I think @michaz even suggested to simply expose the entire map

Reply:
Is it also useful to only expose the shared EncodedValues? So not all, but the ones we need here. Or do we need both (e.g. getAll and getAllShared)?

Reply:
Good question, for the purpose I encountered so far I only need the shared ones, but not sure about other use cases.